src/main/java/org/example/ListNode.java

```diff
@@ -0,0 +1,52 @@
+package org.example;
+
+public class ListNode {
+    int val;
+    ListNode next;
+    ListNode() {}
+    ListNode(int val) { this.val = val; }
+    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
+
+    /**
+     * Added just for testing.
+     */
+    public void printThisAndAllNext()
+    {
```

Comment:
In Java we have this code style. Please configure checkstyle in your IDE and pom.xml
```suggestion
    public void printThisAndAllNext() {
```